pulp_python/app/models.py

```diff
@@ -88,12 +89,13 @@ class PythonPackageContent(Content):
     """
 
     TYPE = 'python'
+    repo_key_fields = ("filename",)
     # Required metadata
-    filename = models.TextField(unique=True, db_index=True)
+    filename = models.TextField(db_index=True)
     packagetype = models.TextField(choices=PACKAGE_TYPES)
     name = models.TextField()
     version = models.TextField()
-    sha256 = models.CharField(max_length=64)
+    sha256 = models.CharField(unique=True, db_index=True, max_length=64)
     # Optional metadata
```

Comment:
Just to make it crystal clear what the constraint is, let's use `unique_together = ('sha256', )`

Reply:
I added the `unique=True` because of how I generated the migration. I couldn't get the auto-generated migration to actually migrate when changing the `unique_together` field, it would always give an error. `unique_together` is needed for the `natural_key_fields` method so I added it back, but I had to use the modify django state operation to stop django from thinking there is more changes than there is.

Reply:
Ah, frustrating.  I think I have hit that issue on occasion as well.  In that case, it does no harm, so I have no objection to leaving it.